include/mumble/Base64.hpp

```diff
@@ -3,30 +3,134 @@
 // that can be found in the LICENSE file at the root of the
 // Mumble source tree or at <https://www.mumble.info/LICENSE>.
 
+// Inspired by https://renenyffenegger.ch/notes/development/Base64/Encoding-and-decoding-base-64-with-cpp.
+
 #ifndef MUMBLE_BASE64_HPP
 #define MUMBLE_BASE64_HPP
 
-#include "Macros.hpp"
-#include "NonCopyable.hpp"
 #include "Types.hpp"
 
-#include <memory>
+#include <array>
+#include <cmath>
+#include <limits>
+
+#include <gsl/span>
 
 namespace mumble {
-class MUMBLE_EXPORT Base64 : NonCopyable {
+class Base64 {
 public:
-	class P;
+	using StrView      = gsl::span< char >;
+	using StrViewConst = gsl::span< const char >;
+
+	static constexpr size_t decode(const BufView out, StrViewConst in) {
```

Comment:
Making an output parameter `const` seems wrong.

Furthermore, we should add an overload that will return e.g. a `std::vector` containing the decoded data. It can internally simply call this function, but API-wise that'll be more straight forward as long as you don't need a specific container for the out data.

Reply:
> Making an output parameter `const` seems wrong.

It's the view itself to be `const`, not the actual buffer (that would be `BufViewConst`).

> Furthermore, we should add an overload that will return e.g. a `std::vector` containing the decoded data. It can internally simply call this function, but API-wise that'll be more straight forward as long as you don't need a specific container for the out data.

Yes, we will definitely add overloads/helpers in the future.

Reply:
> It's the view itself to be const, not the actual buffer (that would be BufViewConst).

I know. That's why I said "seems wrong" and not "is a bug" :)
It is still semantically weird.

Reply:
Yeah, I can see why.